# Assignment: Testing the "Average Age" Claim

#### Task 1: Hypotheses

H<sub>0</sub>: The average age is 30 years old.

H<sub>A</sub>: The average age <i>is not</i> 30 years old.

#### Task 2: Z-Test

In [73]:
import numpy as np
from statsmodels.stats.weightstats import ztest

# 1. Sample Data (Now using a larger sample for Z-test validity)
# Note: Z-tests are more reliable with n > 30
ages = np.array([28, 32, 35, 29, 31, 34, 30, 27, 33, 31, 
                 29, 30, 32, 28, 30, 31, 34, 29, 30, 32,
                 33, 27, 28, 31, 30, 32, 29, 30, 34, 31])

# 2. Set the Population Mean to test against (Null Hypothesis)
pop_mean_null = 30

# 3. Perform One-Sample Z-Test
# value: the mean under the null hypothesis
z_stat, p_val = ztest(ages, value=pop_mean_null)

# 4. Print Results
print(f"Z-Statistic: {z_stat:.4f}")
print(f"P-Value: {p_val:.4f}")

# 5. Decision Logic
alpha = 0.05
if p_val < alpha:
    print(f"Decision: Reject H0 (p < {alpha})")
else:
   print(f"Decision: Fail to Reject H0 (p >= {alpha})") 

print(f"Mean: {sum(ages)/len(ages)}")

Z-Statistic: 1.7072
P-Value: 0.0878
Decision: Fail to Reject H0 (p >= 0.05)
Mean: 30.666666666666668


#### Task 3: Outliers

In [72]:
ages = np.array([28, 32, 35, 29, 31, 34, 30, 27, 33, 31, 
                 29, 30, 32, 28, 30, 31, 34, 29, 30, 32,
                 33, 27, 28, 31, 30, 32, 29, 30, 34, 31,
                 110 ## Added age 110 to dataset for Task 3
                 ])

## Rerunning z-test
z_stat, p_val = ztest(ages, value=pop_mean_null)

## Printing results
print(f"Z-Statistic: {z_stat:.4f}")
print(f"P-Value: {p_val:.4f}")

## Decision logic
alpha = 0.05
if p_val < alpha:
    print(f"Decision: Reject H0 (p < {alpha})")
else:
   print(f"Decision: Fail to Reject H0 (p >= {alpha})") 

print(f"Mean: {sum(ages)/len(ages)}")

Z-Statistic: 1.2470
P-Value: 0.2124
Decision: Fail to Reject H0 (p >= 0.05)
Mean: 33.225806451612904


This single data point decreased the z-statistic by approximately 0.5 and incresed the p-value by about 0.125. The p-value remains statistically insignificant, meaning that the average age must be very close to, if not 30 years of age even when an incredible outlier is added to the dataset. Even though this is what the z-statistic claims, it is technically inaccurate, as the mean is now 33.2, meaning that the z-test is not well equipped to deal with significant outliers.

#### Task 4: Sample Size

If we kept the same sample mean but quadrupled the sample size, the z-statistic and p-value would become more accurate. The z-score would double but the p-value would decrease massively, as seen through the adjusted dataset below (quadrupled sample size). The larger sample size makes it 'easier' to reject the null hypothesis because it means that what occurs is less likely due to random chance. The null hypothesis is rejected because the mean is not exactly equal to 30, so the null hypothesis is rejected.

In [74]:
ages = np.array([28, 32, 35, 29, 31, 34, 30, 27, 33, 31, 
                 29, 30, 32, 28, 30, 31, 34, 29, 30, 32,
                 33, 27, 28, 31, 30, 32, 29, 30, 34, 31,
                 28, 32, 35, 29, 31, 34, 30, 27, 33, 31, 
                 29, 30, 32, 28, 30, 31, 34, 29, 30, 32,
                 33, 27, 28, 31, 30, 32, 29, 30, 34, 31,
                 28, 32, 35, 29, 31, 34, 30, 27, 33, 31, 
                 29, 30, 32, 28, 30, 31, 34, 29, 30, 32,
                 33, 27, 28, 31, 30, 32, 29, 30, 34, 31,
                 28, 32, 35, 29, 31, 34, 30, 27, 33, 31, 
                 29, 30, 32, 28, 30, 31, 34, 29, 30, 32,
                 33, 27, 28, 31, 30, 32, 29, 30, 34, 31
                 ])

## Rerunning z-test
z_stat, p_val = ztest(ages, value=pop_mean_null)

## Printing results
print(f"Z-Statistic: {z_stat:.4f}")
print(f"P-Value: {p_val:.4f}")

## Decision logic
alpha = 0.05
if p_val < alpha:
    print(f"Decision: Reject H0 (p < {alpha})")
else:
   print(f"Decision: Fail to Reject H0 (p >= {alpha})") 

print(f"Mean: {sum(ages)/len(ages)}")

Z-Statistic: 3.4583
P-Value: 0.0005
Decision: Reject H0 (p < 0.05)
Mean: 30.666666666666668


#### Task 5: Size Effect

With a p-value of 0.04, rejecting the null hypothesis, and sample mean being 30.5 years, everything being tested was statistically true. The mean was not exactly 30, it was 30.5. However, in the context of the community center this does not truly matter; this finding is not practically significant for them. The community center would most likely not care about incredibly precise ages. The difference between someone who is thirty and a half vs someone who just turned thirty is insignificant to the community center, they are both thirty year-olds who will engage in similiar activities. This means that even though something is statistically accurate, it is not necessarily contextually releveant.